### ISSUE TITLE
Connection Problems with version 3.6.x

### ISSUE THREAD

**Issue by guidfort** [jfrog/jenkins-artifactory-plugin#265]
If I call "Test Connection" with version 3.6.x then I get the error message "Error occurred while requesting version information: null". With version 3.5.0 no problems.

I see the following differences:

In version 3.5.0 the "password:" is empty in 3.6.x not more

**3.6.x**
artifactoryUrl: https://my-artifact.de:443/artifactory
artifactory.timeout: 300
artifactory.bypassProxy: false
useCredentialsPlugin: true
credentialsId: my_pwd
username: 
password: {Axxxxxxxxo/YUyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyQ0=}
connectionRetry: 3

**3.5.0**
artifactoryUrl: https://my-artifact.de:443/artifactory
artifactory.timeout: 300
artifactory.bypassProxy: false
useCredentialsPlugin: true
credentialsId: my_pwd
username: 
password: 
connectionRetry: 3

**Comment by mbush78** [comment]
This same message just started showing up in one of our Jenkins instances. However two other instance we have the plugin is working fine and a version number is received.

**Comment by guidfort** [comment]
Can I test something else?

**Comment by yahavi** [comment]
Thanks for reporting this issue, @guidfort!
We'll soon start looking at it.

This is related to a security issue we fixed in the last version.
is it an issue only with the "Test Connection" button? Does running jobs still works for you?

**Comment by guidfort** [comment]
The problem was the build jobs. With "Test Connection" I was only able to isolate the problem.

**Comment by guidfort** [comment]
What is the current status?

**Comment by naiztruk96** [comment]
I'm currently on version 3.7 of the plugin which incorporates JFrog Pipelines now. Getting the following error message:

Test ConnectionError occurred while requesting version information: Unexpected character ('<' (code 60)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')at [Source: (org.apache.http.client.entity.LazyDecompressingInputStream); line: 1, column: 2]

There's no clear issue documented on the vendor's (JFrog or Jenkins) Confluence pages for this yet. Please, if anyone has found a workaround here other than a rollback keep us posted.

**Comment by yahavi** [comment]
@naiztruk96, did you change the URL lastly?
It is a symptom of an HTTP request that return an HTML page instead of JSON. It looks like the URL is not pointing to the API URL, but to the UI.
Please make sure that your Artifactory URL ends with `/artifactory`.
For example: 
`https://jfrog-platform-url/artifactory`
See documentation here: https://www.jfrog.com/confluence/display/JFROG/Configuring+Jenkins+Artifactory+Plug-in#ConfiguringJenkinsArtifactoryPlug-in-ArtifactoryPluginConfiguration.

Please let us know if it helped.

**Comment by mbush78** [comment]
I have this plugin in 4 different systems. All configured with the exact same URL. It started with just 1 system giving this 'null' message when you check Test Connection. then out of no where a few weeks later it stopped working on two other systems. It's now only working in one environment of mine.

**Comment by eyalbe4** [comment]
This is indeed strange @mbush78.
Let's try to get to the bottom of this. When clicking on the *Test connection* button, Jenkins sends to Artifactory the following GET request:
```
curl -u<username>:<password>-XGET https://<base url>/artifactory/api/system/version
```
Can you please try replacing the username, password and base URL values and try running it? Note that in the Jenkins UI, the URL configured should be [base url]/artifactory (with the /artifactory suffix), unless your reverse proxy tales care of adding it.
Also, please make sure to add the exact credentials configured in Jenkins.
Let us know what you get.

**Comment by mbush78** [comment]
by chance did this plugin previously have the ability to allow for insecure certs?
Your curl command as is fails when I test it because our Artifactory uses self signed certs. 
If I add the -k option to the curl command I get back expected results.

**Comment by eyalbe4** [comment]
I see @mbush78. That's a good hint we got.
The Artifactory Plugin never supported bypassing SSL certificates (this may a nice feature though). Could it be that your Jenkins instance does not have your Artifactory domain's SSL certificate set up?

**Comment by mbush78** [comment]
Well, we add the cert to the instance of Java that's used for Jenkins which has always worked in the past for other areas the have issues with the self signed cert.  But curl by itself wouldn't be using the Java CA's does it? Am I going to have to explicitly declare in Jenkins startup to use the truststore it already uses anyway because its being run by Java just so curl will see it?  I'm just a lil confused how it could just stop working if nothing was changed as far as SSL goes.   I'll poke around here some more and try running in various ways I can think of.  .... Presuming of course the cert issue is actually the issue since the only message returned is "null"

**Comment by eyalbe4** [comment]
@mbush78,
You're right. Jenkins just uses the java ca cert and curl doesn't.
Another way to prove that this indeed a certificate issue is by putting an http proxy (like postman or charles) and see the response returned to Jenkins after clicking the *Test connection* button. You can also look at Artifactory's *requests.log* and see if the request actually gets to Artifactory. If this is a certificate issue, then the request should get blocked by the reverse proxy and not get to Artifactory at all.
Hopefully in the future we will have an "insecure TLS" option next to the *Test connection* button. This can really help with troubleshooting.
Please share with us your findings and if we can assist in any way,

**Comment by guidfort** [comment]
I also used curl here once.
`With curl -u<username>:<password>-XGET https://<base url>/artifactory/api/system/version `
I have a problem.

I also have to use
`curl **-k** -u<username>:<password>-XGET https://<base url>/artifactory/api/system/version `
This gives me a result

**Comment by eyalbe4** [comment]
https://github.com/jfrog/jenkins-artifactory-plugin/pull/303 fixes another issue which may contribute to the issues described here.
We intend to release a patch which includes https://github.com/jfrog/jenkins-artifactory-plugin/pull/303 and #300 soon.

**Comment by eyalbe4** [comment]
Version 3.7.2 of the Jenkins Artifactory Plugin has been released.
This patch addresses this issue as well.
We would love to get your feedback for it.
Thanks!

**Comment by guidfort** [comment]
I tested with Plugin 3.7.3 and Jenkins 2.244 and I still have the mistak.

**Comment by eyz2020** [comment]
Still have the issue with 3.7.3, Any update to address this yet?

**Comment by guidfort** [comment]
With 3.7.3 still the problem. I saw there is a version 3.8.0. I have not tested this. I am currently having update problems. I suddenly get "sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target"

**Comment by guidfort** [comment]
I could fix my update problem. I check with version 3.8.0 have same problem, with button "Test connection" I got the "null" exception see above.

**Comment by eyz2020** [comment]
Yes the above error is related with the root cert java validation, i have fixed it adding the root certs to the local java.
3.8.0 have the same null exception issue though.

**Comment by guidfort** [comment]
Good news I could fix my "Test Connection" problem. I had to activate "Bypass HTTP proxy". Java debugging and open source is great!!!

**Comment by amanbisht** [comment]
I'm still facing the above issue even after checking the Bypass HTTP proxy setting. I'm using 3.10.0 version.

**Comment by RobiNino** [comment]
Since using "Bypass HTTP proxy" is only a workaround, I'm reopening this issue.
This should be resolved once [jfrog/build-info#421](https://github.com/jfrog/build-info/pull/421) is merged and released.

**Comment by RobiNino** [comment]
Hi @guidfort @mbush78 @eyz2020 @amanbisht ,
Thanks for reporting this issue.
A fix has been merged and will be included in the plugin's upcoming release.

Here's a snapshot version of the plugin that includes the fix:
[https://oss.jfrog.org/oss-snapshot-local/org/jenkins-ci/plugins/artifactory/3.10.x-SNAPSHOT/artifactory-3.10.x-20201220.132410-5.hpi](https://oss.jfrog.org/oss-snapshot-local/org/jenkins-ci/plugins/artifactory/3.10.x-SNAPSHOT/artifactory-3.10.x-20201220.132410-5.hpi).

We'd appreciate your feedback on that.

**Comment by amanbisht** [comment]
@RobiNino When would this plugin update would get released? We have an upcoming and we are dependent on this.

**Comment by yahavi** [comment]
@amanbisht,
We are planning to release the next version by the end of this week, but hopefully sooner.

**Comment by RobiNino** [comment]
Hi @guidfort @mbush78 @eyz2020 @amanbisht ,
Jenkins Artifactory plugin 3.10.1 is released. This release includes a fix for this issue. Feel free to upgrade.
We'd appreciate your feedback on that.

**Comment by guidfort** [comment]
So I tested with 3.10.3 and in both case the plugin (with "Bypass ..." active/inactive) works!
